sklearn/manifold/t_sne.py

```diff
@@ -423,18 +423,19 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
         dist_X = X
     else:
         dist_X = pairwise_distances(X, squared=True)
-    ind_X = np.argsort(dist_X, axis=1)
+    np.fill_diagonal(dist_X, np.inf)  # excludes dist(x_i, x_i)
+    ind_X = np.argsort(dist_X, axis=1)[:, :-1]
     ind_X_embedded = NearestNeighbors(n_neighbors).fit(X_embedded).kneighbors(
         return_distance=False)
 
     n_samples = X.shape[0]
-    t = 0.0
-    ranks = np.zeros(n_neighbors)
-    for i in range(n_samples):
-        for j in range(n_neighbors):
-            ranks[j] = np.where(ind_X[i] == ind_X_embedded[i, j])[0][0]
-        ranks -= n_neighbors
-        t += np.sum(ranks[ranks > 0])
+    inverted_index = np.zeros((n_samples, n_samples))
```

Comment:
silly question but shouldn't this be sparse? I guess depends on whether n_samples >> n_neighbors?

Reply:
We're storing dense data in it.

The O(n^2) isn't great, but it's still an improvement, isn't it?

Reply:
oh, I guess I didn't understand what's going on. Maybe some comments would be nice ;)

Reply:
Sure, I just added some.